testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -3895,31 +3914,65 @@ fn forked_tenure_is_ignored() {
         thread::sleep(Duration::from_secs(1));
     }
 
-    info!("Tenure B broadcasted a block. Issue the next bitcon block and unstall block commits.");
-    let block_tenure_b = NakamotoChainState::get_canonical_block_header(chainstate.db(), &sortdb)
+    sleep_ms(1000);
```

Comment:
Rather than a bunch of `sleep_ms` invocations, this should `wait_for()` on a mutex or `get_chain_info` result, etc. for exactly the conditions that the `sleep_ms` invocations are attempting to wait for. This will make things more resilient to timing issues across computers, CI, etc, but also make the tests more explicit about the behavior they are waiting on.